.github/workflows/_swift.yml

```diff
@@ -5,172 +5,79 @@ on:
 
 jobs:
   build:
-    name: build-${{ matrix.sdk }}
-    runs-on: ${{ matrix.runs-on }}
+    name: ${{ matrix.job_name }}
+    runs-on: macos-15
+    permissions:
+      contents: read
+      id-token: "write"
+    env:
+      XCODE_VERSION: "16.2"
     strategy:
       fail-fast: false
       matrix:
         include:
-          - sdk: macosx
-            runs-on: macos-14
-            platform: macOS
-            xcode: "15.2"
-            destination: platform=macOS
-          - sdk: iphoneos
-            runs-on: macos-14
-            platform: iOS
-            xcode: "15.2"
-            destination: generic/platform=iOS
-    permissions:
-      contents: read
-      id-token: "write"
-    defaults:
-      run:
-        working-directory: ./swift/apple
+          - job_name: build-ios
+            rust-targets: aarch64-apple-ios
+            build-script: scripts/build/ios-appstore.sh
+            ARTIFACT_PATH: "./Firezone.ipa"
+            upload-script: scripts/upload/app-store-connect.sh
+
+          - job_name: build-macos-appstore
+            rust-targets: aarch64-apple-darwin x86_64-apple-darwin
+            build-script: scripts/build/macos-appstore.sh
+            ARTIFACT_PATH: "./Firezone.pkg"
+            upload-script: scripts/upload/app-store-connect.sh
+
+          - job_name: build-macos-standalone
+            rust-targets: aarch64-apple-darwin x86_64-apple-darwin
+            build-script: scripts/build/macos-standalone.sh
+            # mark:next-apple-version
+            ARTIFACT_PATH: "./firezone-macos-client-1.4.0.dmg"
+            NOTARIZE: "${{ (github.event_name == 'workflow_dispatch' || github.ref == 'refs/heads/main') && 'true' || 'false' }}"
+            # mark:next-apple-version
+            upload-script: scripts/upload/github.sh
+            RELEASE_NAME: macos-client-1.4.0
```

Comment:
This was refactored to be a matrix for each distributable, with the upload process carved out into another script.

secrets are not available from the `strategy` context, otherwise I would have added them here and not in the build steps below.